File: week01/day07/frontend/src/types/checkin.ts
export interface CheckinRecord {
  id: number;
  worker_id: number;
  patient_id: number;
  schedule_id: number | null;
  start_time: string;
  end_time: string | null;
  content: string | null;
  status: 'started' | 'completed' | 'absent';
  is_makeup: boolean;
  created_at: string;
  patient_name?: string;
  worker_name?: string;
}

export interface CheckinStartData {
  schedule_id: number;
}

export interface CheckinSubmitData {
  content: string;
}

export interface CheckinMakeupData {
  patient_id: number;
  start_time: string;
  end_time: string;
  content: string;
}

File: week01/day07/backend/schemas/checkin.py
from pydantic import BaseModel, Field


class CheckinStart(BaseModel):
    schedule_id: int = Field(..., ge=1)


class CheckinSubmit(BaseModel):
    content: str = Field(..., min_length=1, max_length=2000)


class CheckinMakeup(BaseModel):
    patient_id: int = Field(..., ge=1)
    start_time: str  # ISO 8601
    end_time: str    # ISO 8601
    content: str = Field(..., min_length=1, max_length=2000)


class CheckinOut(BaseModel):
    id: int
    worker_id: int
    patient_id: int
    schedule_id: int | None = None
    start_time: str
    end_time: str | None = None
    content: str | None = None
    status: str
    is_makeup: bool
    created_at: str
    patient_name: str | None = None
    worker_name: str | None = None

    model_config = {"from_attributes": True}
